scripts/generate_refcats_gen3.py

```diff
@@ -143,75 +140,40 @@ def _make_range(start, end):
 
 DEST_DIR = "${AP_VERIFY_CI_HITS2015_DIR}/preloaded/"
 STD_REFCAT = "refcats"
-DEST_RUN = "refcats/imported"
-
-
-def _rename_dataset_type(type, name):
-    """Create a DatasetType that differs from an existing one in name only.
-
-    Parameters
-    ---------
-    type : `lsst.daf.butler.DatasetType`
-        The type to rename.
-    name : `str`
-        The new name to adopt.
-
-    Returns
-    -------
-    new_type : `lsst.daf.butler.DatasetType`
-        The new DatasetType.
-    """
-    return DatasetType(name, type.dimensions, type.storageClass, type.parentStorageClass)
-
 
 src_repo = Butler(args.src_dir, collections=args.src_collection, writeable=False)
-dest_repo = Butler(DEST_DIR, run=DEST_RUN, writeable=True)
+dest_repo = Butler(DEST_DIR, writeable=True)
 
 
-def _remove_refcat_run(butler, run):
-    """Remove a refcat run and any references from a repository.
+def _remove_refcat_runs(butler):
+    """Remove any old refcat runs from a repository.
 
     Parameters
     ----------
     butler : `lsst.daf.butler.Butler`
         The repository from which to remove ``run``.
-    run : `str`
-        The run to remove, if it exists.
     """
     try:
-        refcat_runs = butler.registry.getCollectionChain(STD_REFCAT)
-        if run in refcat_runs:
-            new_runs = list(refcat_runs)
-            new_runs.remove(run)
-            butler.registry.setCollectionChain(STD_REFCAT, new_runs)
+        refcat_runs = list(butler.registry.getCollectionChain(STD_REFCAT))
     except (lsst.daf.butler.registry.MissingCollectionError, TypeError):
-        pass  # No STD_REFCAT chain; nothing to do
+        return  # No STD_REFCAT chain; nothing to do
 
-    try:
-        butler.removeRuns([run], unstore=True)
-    except lsst.daf.butler.registry.MissingCollectionError:
-        pass  # Already removed; nothing to do
+    butler.registry.setCollectionChain(STD_REFCAT, [])
+
+    butler.removeRuns(refcat_runs, unstore=True)
 
 
 logging.info("Preparing destination repository %s...", DEST_DIR)
-_remove_refcat_run(dest_repo, DEST_RUN)
-dest_repo.registry.registerCollection(DEST_RUN, CollectionType.RUN)
-for src_cat, dest_cat in REFCATS.items():
-    src_type = src_repo.registry.getDatasetType(src_cat)
-    dest_type = _rename_dataset_type(src_type, dest_cat)
-    dest_repo.registry.registerDatasetType(dest_type)
+_remove_refcat_runs(dest_repo)
 dest_repo.registry.refresh()
 
 logging.info("Searching for refcats in %s:%s...", args.src_dir, args.src_collection)
 query = f"htm{HTM_LEVEL} in ({','.join(id_ranges)})"
```

Comment:
I believe we can now support this directly in the bind parameter.

```python
datasets = src_repo.register.queryDatasets(REFCATS, where=f"htm{HTM_LEVEL} in (levels)", bind={"levels": id_ranges}, findFirst=True)
```

Reply:
Yes, I'm saying that in the past year we modified the bind system to correctly handle `IN` with an iterable. I think it was done in d145c3acf0761926c270e138d04f40a9677f2640 (DM-36325).

Reply:
Opened [DM-38916](https://jira.lsstcorp.org/browse/DM-38916) to add this feature to the docstring (it's already documented in topic pages).

Reply:
Hmm... it looks like I can't use `bind` in this case unless I give up the range compression done by `_make_range`, and I feel a bit nervous doing that.

```
lsst.daf.butler.registry.queries.expressions._predicate.ExpressionTypeError: Invalid type <class 'str'> for element in <class 'int'> IN expression 'htm7 IN (shards)'.
```

Reply:
Oh, because you've added strings in `n..m` syntax already? Okay. Yes, we are expecting just the integers.